File: 0169-majority-element/0169-majority-element.go
func majorityElement(nums []int) int {
    candidate := 0
    count := 0
    
    // https://www.geeksforgeeks.org/boyer-moore-majority-voting-algorithm/
    
    for _, num := range nums {
        if count == 0 {
            candidate = num
        }
        if num == candidate {
            count++
        } else {
            count --
        }
    }
    
    return candidate 
}

File: 0169-majority-element/0169-majority-element.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        hashmap = {}
        
        for num in nums:
            if num in hashmap:
                hashmap[num]+=1
            else :
                hashmap[num] = 1
        
        mx = 0
        
        for key, value in hashmap.items():
            if value > mx:
                mx = value
                majority_element = key
            
        return majority_element